### ISSUE TITLE
Recon suggest widget: enable middle-click to open URL in new tab without selecting

### ISSUE THREAD

**Issue by wetneb** [OpenRefine/OpenRefine#6697]
When using the recon suggest widget in some field (such as manually selecting a recon candidate), it would be helpful if we could visit the page for the recon candidate without selecting it for matching.

### Proposed solution

Enable the "middle click" on suggested entities so that their page opens in a new tab.

### Alternatives considered

Not sure?

### Additional context

This was prompted by a request from adem özcan from the OpenRefine-Wikimedia Telegram channel:

> There is no short description written in the suggestion in this image. Sometimes the short description is not enough, I want to look at the page to be sure. How can I make the page open in a new tab?
> 
> ![image](https://github.com/OpenRefine/OpenRefine/assets/309908/88662835-50c0-4de2-a9c7-4254760379c7)

They confirmed that the middle-click option would solve their problem.

**Comment by antoine2711** [comment]
There is no middle click on Mac…

Why no use the contextual menu?

Regards, Antoine

**Comment by wetneb** [comment]
> Why no use the contextual menu?

what do you mean with that? What would it look like?

**Comment by antoine2711** [comment]
> what do you mean with that? What would it look like?

@wetneb : with this code, right click (contextual menu) on the div will open a new tab with the URL to the Wikidata page.
 
![image](https://github.com/OpenRefine/OpenRefine/assets/12217525/8d66cd32-ca3c-464c-bf95-0f0a6fd95d3a)

Regards, Antoine

**Comment by wetneb** [comment]
Ah, I would have understood it better if you had written "right click" instead of "contextual menu" since your solution precisely does not involve any contextual menu ;)

I think in general it's not a great practice to block access to the contextual menu (even if there are ways to recover it), so I'd be a bit reluctant to use that. Also, it's not very well discoverable.

On systems which have it, the middle click can be used to open a link in a new tab. Is there any shortcut you can use on Mac OS to open a link in a new tab instead of in the current tab? A click with a modifier key enabled? I think it would make sense to match that, since that's semantically what we are trying to accomplish.

**Comment by antoine2711** [comment]
> On systems which have it

Mac never had middle click. Mouses on the Mac always had only 2 buttons. When Macs only have one button (that was Steve Jobs mentallity: one button is enough, more cause confusion and complexity), the control key with the mouse click equals to the right click. But I’ve never heard about a key for the middle click.

Blocking the contextual menu inside a menu is not a big problem, a menu shouldn’t be called inside another menu.

For the fact that it’s not discoverable, we could have text at the bottom of the menu that give indication about this functionalty.

But, more simply, when could just make the items name (Canada in my example, Buyukbayat in yours) a link, and it would have a _blank target, so open in another tab. That would be both simple and very normal.

Regards, Antoine

**Comment by wetneb** [comment]
After a quick search, it looks that "Command-Click" is the standard way to open a page in a new tab on Mac OS.
So I would propose to enable both the middle-click and Command-click to open the suggest menu items in new tabs.

**Comment by wetneb** [comment]
Marking this as a good first issue, with the goal that:
* on Linux and Windows, the middle click button opens the URL associated with a suggested item without selecting it
* on MacOS, Command + Click does the same

**Comment by 2897robo** [comment]
Hi! 👋

I'd love to contribute to this issue. The functionality seems well-scoped and would be a great first contribution. May I take this on?

Also, I plan to support:
- Middle-click to open the entity's link in a new tab (for Linux/Windows)
- Command-click for macOS users

Let me know if this approach sounds good!

**Comment by SoryRawyer** [comment]
Hey @2897robo, I just wanted to check in and see how this is going. Do you have any questions about the feature, the implementation, or the contribution process?

**Comment by 2897robo** [comment]
Hi @SoryRawyer,

Thanks for checking in! After initially exploring the issue, I realized that the implementation might be a bit more complex than I first expected, especially regarding event handling differences across platforms.

I’ve decided to step back from this one for now, but I really appreciate the opportunity to take a look. I’d love to try contributing to another issue in the future that’s a better fit for my current skill level.

Thanks again for the support!

**Comment by SoryRawyer** [comment]
Thanks for the heads up, @2897robo! Please feel free to reach out in the future with any questions about getting started on this or another issue!

**Comment by riyaseema80** [comment]
Hi 
I’d like to work on this issue.  
I’ll try to implement middle-click to open suggestion links in a new tab, and will test carefully.  
Please assign to me if that is okay.

**Comment by SoryRawyer** [comment]
Hi @riyaseema80, thanks for your interest! I'll go ahead and assign this to you. Please feel free to reach out either here or on [the forum](https://forum.openrefine.org/) if you want to talk though anything.

**Comment by riyaseema80** [comment]
Thanks for reviewing and merging! Happy to contribute more.

**Comment by SoryRawyer** [comment]
@riyaseema80 I think you may have opened the pull request against your fork and not the main OpenRefine repository. Could you please reopen your pull request in this repository?

**Comment by riyaseema80** [comment]
@SoryRawyer  Thanks for the clarification!
I realized the mistake — the original PR was indeed opened on my fork.
I’ve now opened the PR correctly on the main OpenRefine repo.

**Comment by hariprabhu571** [comment]
Hi @SoryRawyer,  I would like to work on this issue. 
Could you please assign it to me?

**Comment by Sujithrababu** [comment]
Hi @SoryRawyer, I’d like to work on this issue. 
Is it currently available to be assigned?

**Comment by hasanefeuygun** [comment]
Hi , I noticed that PR #7793 is currently waiting on the requested Cypress end-to-end test.I'd be happy to pick this up and complete the missing test so we can move issue #6697 forward.Would you prefer taht I contribute to the existing PR, or open a replacement PR based on the current implementation?

**Comment by tfmorris** [comment]
Hi @hasanefeuygun. Welcome to the project. I've assigned you to this issue. I don't think you can modify someone else's PR, but feel free to pick up that branch and build on it for your own PR. Thanks!

**Comment by hasanefeuygun** [comment]
Hi, I’ve opened #7944 for this issue.

The PR adds support for:
- middle-click to open a reconciliation suggestion in a new tab without selecting it
- Cmd+click on macOS for the same behavior

It also includes Cypress regression coverage and preserves the existing normal-click selection behavior.

I verified both modified-click interactions manually on macOS as well.
